### ISSUE TITLE
Using truethy coroutine in if-clause should trigger a warning

### ISSUE THREAD

**Issue by hf-kklein** [pylint-dev/pylint#9502]
### Current problem

```python
async def is_ok() -> bool:
    return False


async def my_code():
    ok = is_ok()  # <-- it's not awaited!
    if ok:  # a warning here would be nice
        print("Should never go here")  # <-- we end up here
    else:
        print("Fine")

```

There's a **runtime** warning:
> Lib\asyncio\events.py:80: RuntimeWarning: coroutine 'is_ok' was never awaited

But I'm missing a linter warning.
The code would run into the `ok` block, because the coroutine is truethy (although not `True`).

### Desired solution

My feature request is:

Using
```python
if something_unawaited_with_boolean_return_value:
```
should trigger a pylint warning.

so with above example:
```python
if is_ok():
```
leads to a warning but
```python
if is_ok() is True:
```
is fine.

### Additional context

This related to #8870 but this feature request is "narrower" / more fine grained.
Personally, I think it's generally ok, to not await stuff directly.

```python
async def foo()->Any:
    pass

async def bar()->Any:
   pass

x = foo() # I wouldn't want a pylint warning here. How does pylint know what I'm gonna do with x?
y = bar()
await asyncio.gather(x, y)
```

But the use in an if-clause is very error-prone imho.

**Comment by Pierre-Sassoulas** [comment]
Added the discussion label because I feel that a type checker would have a easier time with this issue.

**Comment by DanielNoord** [comment]
Indeed, `mypy` already has this feature. I would lean towards delegating this to them :smile:

**Comment by DanielNoord** [comment]
I think you need to enable it with [truthy-bool](https://mypy.readthedocs.io/en/stable/error_code_list2.html#check-that-expression-is-not-implicitly-true-in-boolean-context-truthy-bool).

**Comment by hf-kklein** [comment]
nice, that's fine enough for me. thanks @DanielNoord !
